docs/dev/metrics.md

```diff
@@ -0,0 +1,56 @@
+# CAO Metrics
+
+The Cluster Autoscaler Operator reports the following metrics:
+
+## Metrics provided by the controller runtime
+
+The [controller runtime](https://github.com/kubernetes-sigs/controller-runtime)
+integration with the operator provides several metrics about the internals of the
+operator. You can find more information about these metrics names and their labels through
+the following links:
+
+### Kubernetes controller metrics
+
+These metric names begin with `controller_runtime_reconcile_`. The labels
+`controller="cluster_autoscaler_controller"` and
+`controller="machine_autoscaler_controller"` can be used to refine queries against these metrics.
+* [Controller runtime reconciliation metrics implementation](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/internal/controller/metrics/metrics.go)
+
+### Admission webhook metrics
+
+These metric names begin with `controller_runtime_webhook_`.  The label
+`webhook="/validate-clusterautoscalers"` can be used to refine the queries for these metrics.
+* [Controller runtime webhook metrics implementation](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/webhook/internal/metrics/metrics.go)
+
+### API REST server metrics
+
+These metric names begin with `rest_client_` and `reflector_`. The labels
+`url` and `verb` can be used to refine these metrics further. The `url` label
+should contain the address to the Kubernetes API server, while `verb`
+containts the type of HTTP request (eg `"GET"`).
+* [Controller runtime Prometheus REST server metrics implementation](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/metrics/client_go_adapter.go)
+
+### Prometheus work queue metrics
+
+These metric names begin with `workqueue_`. The labels
+`name="cluster_autoscaler_controller"` and `name="machine_autoscaler_controller"`
+can be used to refine queries against these metrics.
+* [Controller runtime Prometheus work queue metrics implementation](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/metrics/workqueue.go)
+
+## Metrics about the Prometheus collectors
+
+Prometheus provides some default metrics about the internal state
```

Comment:
I'm not clear on what this section means.  Are these metrics collected automatically by virtue of using some prometheus library in the sig-controller-runtime code?

Reply:
Okay, yeah, that's how it works.  Wasn't clear what this was about.

How about an example.
```process_cpu_seconds_total{job="cluster-autoscaler-operator"}```

Reply:
sure, i can add an example. i do  wonder if i shouldn't update the mao doc as well then since it has the same clause in it?

https://github.com/openshift/machine-api-operator/blob/master/docs/dev/metrics.md#metrics-about-the-prometheus-collectors

note that the sample metric in that section is specifically added by the mao, that isn't the case here.